bitnami/thanos/templates/storegateway/service-sharded.yaml

```diff
@@ -0,0 +1,68 @@
+{{- if and .Values.storegateway.enabled .Values.storegateway.sharded.enabled }}
+
+{{- $shards := int 0 }}
+
+{{- if .Values.storegateway.sharded.hashpartitioning.shards }}
+  {{- $shards = int .Values.storegateway.sharded.hashpartitioning.shards }}
+{{- else }}
+  {{- $shards = len .Values.storegateway.sharded.timepartitioning }}
+{{- end }}
+
+{{- range $index, $_ := until $shards }}
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ printf "%s-storegateway-%s" (include "common.names.fullname" $) $index | trunc 63 | trimSuffix "-" }}
+  labels: {{- include "common.labels.standard" $ | nindent 4 }}
+    app.kubernetes.io/component: storegateway
+    {{- if and $.Values.metrics.enabled $.Values.metrics.serviceMonitor.enabled }}
+    prometheus-operator/monitor: 'true'
+    {{- end }}
+    {{- if $.Values.commonLabels }}
+    {{- include "common.tplvalues.render" ( dict "value" $.Values.commonLabels "context" $ ) | nindent 4 }}
+    {{- end }}
+  {{- if $.Values.storegateway.service.annotations }}
+  annotations: {{- include "common.tplvalues.render" (dict "value" $.Values.storegateway.service.annotations "context" $) | nindent 4 }}
+  {{- end }}
+spec:
+  type: {{ $.Values.storegateway.service.type }}
+  {{- if not (empty $.Values.storegateway.sharded.service.clusterIPs) }}
+  clusterIP: {{ index $.Values.storegateway.sharded.service.clusterIPs $index }}
+  {{- end }}
```

Comment:
I guess we can find an issue if the `storegateway.sharded.service.clusterIPs` array length is different from `storegateway.sharded.timepartitioning` value or `storegateway.sharded.hashpartitioning.shards` array length. Sth similar will happen with `loadBalancerIPs`, `nodePorts` or `loadBalancerSourceRanges`.

I would include a new validation in the `thanos.validateValues` macro to show an error when this happens.

- https://github.com/bitnami/charts/blob/master/bitnami/thanos/templates/_helpers.tpl#L184